src/DbUpdateService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ModelRepoBrowser.Crawler;
+
+namespace ModelRepoBrowser;
+
+/// <summary>
+/// Service that periodically updates the Model Repository Database using <see cref="IRepositoryCrawler"/>.
+/// </summary>
+public class DbUpdateService : BackgroundService
+{
+    private readonly IServiceScopeFactory scopeFactory;
+    private readonly ILogger<DbUpdateService> logger;
+    private readonly IConfiguration configuration;
+
+    public TimeOnly PreferedTime { get; set; }
+
+    public DbUpdateService(IServiceScopeFactory scopeFactory, ILogger<DbUpdateService> logger, IConfiguration configuration)
+    {
+        this.scopeFactory = scopeFactory;
+        this.logger = logger;
+        this.configuration = configuration;
+
+        PreferedTime = new TimeOnly(1, 0);
+    }
+
+    private async Task UpdateModelRepoDatabase()
+    {
+        logger.LogInformation("Updating ModelRepoDatabase...");
+
+        if (!Uri.TryCreate(configuration["Crawler:RootRepositoryUri"], UriKind.RelativeOrAbsolute, out var rootUri))
+        {
+            logger.LogError("Unable to parse configuration Crawler:RootRepositoryUri. Database update skipped.");
+            return;
+        }
+
+        try
+        {
+            using (var scope = scopeFactory.CreateScope())
+            {
+                var crawler = scope.ServiceProvider.GetRequiredService<IRepositoryCrawler>();
+                var repositories = await crawler.CrawlModelRepositories(rootUri).ConfigureAwait(false);
+
+                var context = scope.ServiceProvider.GetRequiredService<RepoBrowserContext>();
+                context.Database.BeginTransaction();
+
+                context.Catalogs.RemoveRange(context.Catalogs);
+                context.Models.RemoveRange(context.Models);
+                context.Repositories.RemoveRange(context.Repositories);
+                context.SaveChanges();
+
+                context.Repositories.AddRange(repositories.Values);
+                context.SaveChanges();
+
+                context.Database.CommitTransaction();
+                logger.LogInformation("Updating ModelRepoDatabase complete");
+            }
+        }
+        catch (DbUpdateException ex)
```

Comment:
Q/REQ: Müssen hier nicht noch die möglichen Exceptions des Crawlers abgefangen werden?

Reply:
Der Crawler müsste bereits genug fehlertolerant sein und in jedem Fall ein leeres Dictionary zurückgeben.

Reply:
Muss dieser Fall nicht abgefangen werden, damit wir nicht die DB leeren und danach _nichts_ in die DB schreiben?

Reply:
@danjov  Wie wir im Falle von einem Fehlgeschlagenen / Offline Repository verfahren ist noch zu klären. Ich werde ein Issue erstellen & es @olivergrimm zuweisen.

Reply:
@Philippluca Wir sollten aber trotzdem bei einem leeren Ergebnis des Crawlers abbrechen und die DB nicht leeren. Das sollte das aktuelle Verhalten sein und der Rest bzw. die Details können mit @olivergrimm besprochen werden.